plugins/nf-google/src/main/nextflow/cloud/google/batch/GoogleBatchMachineTypeSelector.groovy

```diff
@@ -63,27 +81,35 @@ class GoogleBatchMachineTypeSelector {
         int memPerVm
     }
 
-    static String bestMachineType(int cpus, int memoryMB, String region, boolean spot, boolean localSSD, List<String> families) {
+    static GoogleBatchCloudinfoMachineSelector INSTANCE() {
+        if (instance != null)
+            return instance
+        return instance = new GoogleBatchCloudinfoMachineSelector()
+    }
+
+    String bestMachineType(int cpus, int memoryMB, String region, boolean spot, boolean localSSD, List<String> families) {
         final machineTypes = getAvailableMachineTypes(region)
+        if (families == null)
+            families = []
 
         // Check if a specific machine type was defined
-        if( families.size() == 1 ) {
+        if (families.size() == 1) {
             final familyOrType = families.get(0)
-            if( familyOrType.contains("custom-") )
+            if (familyOrType.contains("custom-"))
                 return familyOrType
 
-            if (machineTypes.find {it.type == familyOrType })
+            if (machineTypes.find { it.type == familyOrType })
                 return familyOrType
         }
 
-        final memoryGB = Math.ceil(memoryMB / 1024) as int
+        final memoryGB = Math.ceil(memoryMB / 1024.0 as float) as int
 
         // Use only families that can have a local SSD
         if (!families && localSSD)
-            families = DeFAULT_FAMILIES_WITH_SSD
+            families = DEFAULT_FAMILIES_WITH_SSD
 
         // All types are valid if no families are defined, otherwise at least it has to start with one of the given values
-        final matchMachineType = { t -> !families || families.find{t.startsWith(it)} }
+        final matchMachineType = (String t) -> !families || families.find { matchType(it, t) }
```

Comment:
```suggestion
        final matchMachineType = (String t) -> { !families || families.find { matchType(it, t) } }
```

Fascinating this lambda, likely writing in {} makes a bit more intuitive

Reply:
I think I'm mixing Java and Groovy lambda syntax :exploding_head:  

What you purpose will make the lambda return a lambda, isn't it?

May be the Groovy way is...
```
final matchMachineType = { (String t) -> !families || families.find { matchType(it, t) } }
```
??

Reply:
Ok the groovy way is:

```
final matchMachineType = { String t -> !families || families.find { matchType(it, t) } }
```

Reply:
done

Reply:
Groovy supports both Closure and Lambda syntaxes. If I'm not wrong the above should be equivalent to lambda 

```
final matchMachineType = (String t) -> { !families || families.find { matchType(it, t) } }
```